packaging/hudi-flink-bundle/pom.xml

```diff
@@ -84,7 +84,6 @@
                   <include>org.apache.hudi:hudi-sync-common</include>
                   <include>org.apache.hudi:hudi-hadoop-mr</include>
                   <include>org.apache.hudi:hudi-timeline-service</include>
-                  <include>org.apache.hudi:hudi-aws</include>
 
```

Comment:
Flink code `HiveSyncContext` is dependent on `AwsGlueCatalogSyncTool`.

Reply:
Good Point. The idea is to have hudi-aws classes in the hudi-aws bundle. hudi-flink-bundle won't have any hudi-aws classes to avoid the below issue. We will ask users to place both hudi-flink-bundle and hudi-aws-bundle in the Flink/lib. Do you have any concerns with this approach?

Issue when hudi-flink-bundle includes hudi-aws

```
2023-10-07 14:47:03,463 ERROR org.apache.hudi.sink.StreamWriteOperatorCoordinator          [] - Executor executes action [sync hive metadata for instant 20231007144701183] error
java.lang.NoClassDefFoundError: software/amazon/awssdk/services/glue/model/EntityNotFoundException
    at org.apache.hudi.aws.sync.AwsGlueCatalogSyncTool.initSyncClient(AwsGlueCatalogSyncTool.java:52) ~[hudi-flink-bundle.jar:0.13.1-amzn-1]
    at org.apache.hudi.hive.HiveSyncTool.<init>(HiveSyncTool.java:114) ~[hudi-flink-bundle.jar:0.13.1-amzn-1]
```
This issue happens as AwsGlueCatalogSyncTool (hudi-aws module) part of hudi-flink-bundle has not relocated AWS SDK classes whereas the hudi-aws-bundle (which includes AWS SDK) has relocated it.

Reply:
Then what is the suggested way to load the `AwsGlueCatalogSyncTool`, maybe by reflection?

Reply:
Yes, I also think reflection will be a better choice.

1. If the user configures "hive_sync.mode" to "hms", then only hudi-flink-bundle is needed.
2. If the user configures "hive_sync.mode" to "glue", then both hudi-flink-bundle and hudi-aws-bundle is needed. This needs to be documented [here](https://hudi.apache.org/docs/syncing_aws_glue_data_catalog).

I see ReflectionUtils and SyncUtilHelpers have provided ways to easily load the class. I will use them and submit a new revision if you are also fine with this discussed approach.

Reply:
Makes sense to me.

Reply:
Can you help me to understand why `hudi-aws` must be excluded? Is it required to resolve the class conflicts?

Reply:
Yes, the hudi-aws-bundle needs to be used to access the features present in hudi-aws and its aws dependencies.

Reply:
Fine with it, please keep @umehrot2 as noted.